ts/Core/Chart/MapChart.ts

```diff
@@ -198,6 +216,242 @@ namespace MapChart {
     }
 }
 
+addEvent(MapChart, 'addSeriesAsDrilldown', function (e): boolean {
+    const chart = this,
+        {
+            point,
+            options
+        }: {
+            point: MapPoint,
+            options: SeriesOptions
+        } = e as any;
+
+    // stop hovering while drilling down
+    point.series.isDrilling = true;
+    // stop duplicating and overriding animations
+    point.series.options.inactiveOtherPoints = true;
+
+    if (
+        chart.options.drilldown &&
+        chart.options.drilldown.animation &&
+        chart.options.drilldown.mapZooming
+    ) {
+        // hide and disable dataLabels
+        if (point.series.dataLabelsGroup) {
+            point.series.dataLabelsGroup.destroy();
+            delete point.series.dataLabelsGroup;
+        }
+
+        // first zoomTo then crossfade series
+        chart.disableTransform = false;
+
+        let animOptions: boolean | Partial<AnimationOptions> | undefined = {};
+
+        if (chart.options.drilldown) {
+            animOptions = merge({}, chart.options.drilldown.animation);
+        }
```

Comment:
Use the [animObject](https://github.com/highcharts/highcharts/blob/6ed5d3ae8c8fb9f30e78fc5ef16cf664cbf75a4c/ts/Core/Animation/AnimationUtilities.ts#L86-L95) function to avoid type checking. It takes animation options and always returns an object.